lib/matplotlib/widgets.py

```diff
@@ -835,12 +838,19 @@ def _update(self, event):
             self._active_handle = None
             return
 
-        # determine which handle was grabbed
-        handle = self._handles[
-            np.argmin(
+        if self.orientation == "vertical":
+            # determine which handle was grabbed
+            event_handle = np.argmin(
+                np.abs([h.get_ydata()[0] - event.ydata for h in self._handles])
+            )
+            handle = self._handles[event_handle]
+        else:
+            # determine which handle was grabbed
+            event_handle = np.argmin(
                 np.abs([h.get_xdata()[0] - event.xdata for h in self._handles])
             )
```

Comment:
Minor style improvement:
```suggestion
        # determine which handle was grabbed
        if self.orientation == "vertical":
            handle_index = np.argmin(
                np.abs([h.get_ydata()[0] - event.ydata for h in self._handles])
            )
        else:
            handle_index = np.argmin(
                np.abs([h.get_xdata()[0] - event.xdata for h in self._handles])
            )
        handle = self._handles[handle_index]
```

Reply:
Fixed. Let me know if there is anything else.